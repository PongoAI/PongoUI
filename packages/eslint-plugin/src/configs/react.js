/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
