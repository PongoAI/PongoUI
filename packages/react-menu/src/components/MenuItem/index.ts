export * from './MenuItem';
export * from './useMenuItemStyles';
