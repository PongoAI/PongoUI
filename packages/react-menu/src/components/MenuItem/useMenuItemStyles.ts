import { mergeClasses, makeStyles, shorthands } from '@griffel/react';
import type { MenuItemState } from '@fluentui/react-menu';
import { createCustomFocusIndicatorStyle } from '@fluentui/react-tabster';
import { tokens } from '@pongo-ui/react-theme';

export const menuFocusRectColor = `--pongo-menu-item-focus-color`;

const useStyles = makeStyles({
  root: {
    position: 'relative',
    display: 'flex',
    height: '32px',
    ...shorthands.padding('0px', '10px'),
    backgroundColor: tokens.canvasColor,
    fontFamily: tokens.fontFamilyBase,
    fontSize: tokens.fontSize300,
    alignItems: 'center',
    userSelect: 'none',
    color: tokens.textColor,
    ...shorthands.gap('4px'),
  },

  enabled: {
    cursor: 'pointer',
    [menuFocusRectColor]: tokens.textColor,
    ':hover': {
      backgroundColor: tokens.brandForegroundHover,
    },
    ':active': {
      backgroundColor: tokens.brandForegroundPressed,
    },
    ':focus-within': {
      backgroundColor: tokens.brandForegroundPressed,
    },
  },

  disabled: {
    cursor: 'not-allowed',
    [menuFocusRectColor]: tokens.inheritDisabled,
    color: tokens.inheritDisabled,
  },

  content: {
    paddingLeft: '2px',
    paddingRight: '2px',
    backgroundColor: 'transparent',
    flexGrow: 1,
  },

  description: {
    paddingLeft: '2px',
    paddingRight: '2px',

    color: tokens.inherit,
    ':hover': {
      color: tokens.inherit,
    },
    ':focus': {
      color: tokens.inherit,
    },
  },

  focusIndicator: createCustomFocusIndicatorStyle(
    {
      ':after': {
        content: "''",
        position: 'absolute',
        top: '0px',
        right: '0px',
        bottom: '0px',
        left: '0px',
        boxSizing: 'border-box',
        ...shorthands.border('2px', 'solid', `var(${menuFocusRectColor})`),
      },
    },
    { selector: 'focus-within' },
  ),

  icon: {
    width: '20px',
    height: '20px',
    lineHeight: '0',
    fontSize: '20px',
  },
});

export const useMenuItemStyles = (state: MenuItemState): MenuItemState => {
  const styles = useStyles();

  state.root.className = mergeClasses(
    styles.root,
    styles.focusIndicator,
    !state.disabled ? styles.enabled : styles.disabled,
  );

  if (state.content) {
    state.content.className = mergeClasses(styles.content, state.content.className);
  }

  if (state.secondaryContent) {
    state.secondaryContent.className = mergeClasses(styles.description, state.secondaryContent.className);
  }

  if (state.icon) {
    state.icon.className = mergeClasses(styles.icon, state.icon.className);
  }

  if (state.submenuIndicator) {
    state.submenuIndicator.className = mergeClasses(styles.icon, state.submenuIndicator.className);
  }

  return state;
};
