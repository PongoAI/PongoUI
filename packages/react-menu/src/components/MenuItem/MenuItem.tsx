import * as React from 'react';
import { useMenuItem_unstable, renderMenuItem_unstable } from '@fluentui/react-menu';
import type { MenuItemProps } from '@fluentui/react-menu';
import { useMenuItemStyles } from './useMenuItemStyles';
import type { ForwardRefComponent } from '@fluentui/react-utilities';

/**
 *  Used to create an item in the Menu.
 */
export const MenuItem: ForwardRefComponent<MenuItemProps> = React.forwardRef((props, ref) => {
  const state = useMenuItem_unstable(props, ref);

  useMenuItemStyles(state);

  return renderMenuItem_unstable(state);
});

MenuItem.displayName = 'MenuItem';
