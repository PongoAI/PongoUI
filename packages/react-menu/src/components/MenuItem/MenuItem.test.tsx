import React from 'react';
import { MenuItem } from './MenuItem';
import type { MenuItemProps } from '@fluentui/react-menu';
import { render } from '@testing-library/react';
import { isConformant } from '../../common/isConformant';

describe('MenuItem', () => {
  isConformant({
    Component: MenuItem as React.FunctionComponent<MenuItemProps>,
    displayName: 'MenuItem',
    disabledTests: ['component-handles-classname', 'make-styles-overrides-win'],
  });

  it('renders correctly', () => {
    const result = render(<MenuItem>Test Item</MenuItem>);
    expect(result.container).toMatchSnapshot();
  });

  it('renders secondaryContent correctly', () => {
    const result = render(<MenuItem secondaryContent="test">Test Item</MenuItem>);
    expect(result.container).toMatchSnapshot();
  });

  it('renders disabled correctly', () => {
    const result = render(<MenuItem disabled>Test Item</MenuItem>);
    expect(result.container).toMatchSnapshot();
  });
});
