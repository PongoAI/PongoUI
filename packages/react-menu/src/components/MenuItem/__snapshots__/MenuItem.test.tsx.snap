// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem renders correctly 1`] = `
<div>
  <div
    class="___1x7e2fi f10pi13n f22iagw fvfw249 f1nbblvp f10sqwl7 f1ov4xf1 fne32i3 f8qjfyk fk6fouc f1arxnym f122n59 f1xqy1su fc467gg fnmxc1e fynvm4a f2hkw1w fdxsi5y f1i4q40k fa0joat f1yduw8h f1cwgisr f6dtr1x fvk4ntw f1bkt4b4 f16hf5f2 fyyqeim f1qhqcal f1kbdjx9 fw1d893 f14hlsw1 fpuz8dn fbw1f98 f8d6am2 f1gpnaou f11mktvj f1k6fduh fkcovh2 fc73jsq f8mbk1b f3c9x3h"
    role="menuitem"
    tabindex="0"
  >
    <span
      class="___n49lwn0 f3rmtva fqerorx"
    >
      Test Item
    </span>
  </div>
</div>
`;

exports[`MenuItem renders disabled correctly 1`] = `
<div>
  <div
    aria-disabled="true"
    class="___t2c9jg0 f10pi13n f22iagw fvfw249 f1nbblvp f10sqwl7 f1ov4xf1 fne32i3 f8qjfyk fk6fouc f1arxnym f122n59 f1xqy1su favx1wl fnmxc1e fynvm4a f2hkw1w fdxsi5y f1i4q40k fa0joat f1yduw8h f1cwgisr f6dtr1x fvk4ntw f1bkt4b4 f16hf5f2 fyyqeim f1qhqcal f1kbdjx9 fw1d893 f14hlsw1 fpuz8dn fbw1f98 f8d6am2 f1gpnaou f11mktvj fdrzuqr f1038xnb"
    role="menuitem"
    tabindex="0"
  >
    <span
      class="___n49lwn0 f3rmtva fqerorx"
    >
      Test Item
    </span>
  </div>
</div>
`;

exports[`MenuItem renders secondaryContent correctly 1`] = `
<div>
  <div
    class="___1x7e2fi f10pi13n f22iagw fvfw249 f1nbblvp f10sqwl7 f1ov4xf1 fne32i3 f8qjfyk fk6fouc f1arxnym f122n59 f1xqy1su fc467gg fnmxc1e fynvm4a f2hkw1w fdxsi5y f1i4q40k fa0joat f1yduw8h f1cwgisr f6dtr1x fvk4ntw f1bkt4b4 f16hf5f2 fyyqeim f1qhqcal f1kbdjx9 fw1d893 f14hlsw1 fpuz8dn fbw1f98 f8d6am2 f1gpnaou f11mktvj f1k6fduh fkcovh2 fc73jsq f8mbk1b f3c9x3h"
    role="menuitem"
    tabindex="0"
  >
    <span
      class="___n49lwn0 f3rmtva fqerorx"
    >
      Test Item
    </span>
    <span
      class="___omqrbh0 f1ysjbbf f15s2vri f3krum3"
    >
      test
    </span>
  </div>
</div>
`;
